Drop-In/src/main/java/com/braintreepayments/api/DropInClient.java

```diff
@@ -0,0 +1,428 @@
+    }
+
+    Authorization getAuthorization() {
+        return braintreeClient.getAuthorization();
+    }
+
+    void getConfiguration(ConfigurationCallback callback) {
+        braintreeClient.getConfiguration(callback);
+    }
+
+    void sendAnalyticsEvent(String eventFragment) {
+        braintreeClient.sendAnalyticsEvent(eventFragment);
+    }
+
+    void collectDeviceData(FragmentActivity activity, DataCollectorCallback callback) {
+        dataCollector.collectDeviceData(activity, callback);
+    }
+
+    void performThreeDSecureVerification(final FragmentActivity activity, PaymentMethodNonce paymentMethodNonce, final ThreeDSecureResultCallback callback) {
+        final ThreeDSecureRequest threeDSecureRequest = dropInRequest.getThreeDSecureRequest();
+        threeDSecureRequest.setNonce(paymentMethodNonce.getString());
+
+        threeDSecureClient.performVerification(activity, threeDSecureRequest, new ThreeDSecureResultCallback() {
+            @Override
+            public void onResult(@Nullable ThreeDSecureResult threeDSecureResult, @Nullable Exception error) {
+                if (error != null) {
+                    callback.onResult(null, error);
+                    return;
+                }
+                threeDSecureClient.continuePerformVerification(activity, threeDSecureRequest, threeDSecureResult, callback);
+            }
+        });
+    }
+
+    void shouldRequestThreeDSecureVerification(PaymentMethodNonce paymentMethodNonce, final ShouldRequestThreeDSecureVerification callback) {
+        if (paymentMethodCanPerformThreeDSecureVerification(paymentMethodNonce)) {
+            braintreeClient.getConfiguration(new ConfigurationCallback() {
+                @Override
+                public void onResult(@Nullable Configuration configuration, @Nullable Exception error) {
+                    if (configuration == null) {
+                        callback.onResult(false);
+                        return;
+                    }
+
+                    boolean hasAmount = !TextUtils.isEmpty(dropInRequest.getAmount()) ||
+                            (dropInRequest.getThreeDSecureRequest() != null && !TextUtils.isEmpty(dropInRequest.getThreeDSecureRequest().getAmount()));
+
+                    boolean shouldRequestThreeDSecureVerification =
+                            dropInRequest.shouldRequestThreeDSecureVerification()
+                                    && configuration.isThreeDSecureEnabled()
+                                    && hasAmount;
+                    callback.onResult(shouldRequestThreeDSecureVerification);
+                }
+            });
+
+        } else {
+            callback.onResult(false);
+        }
+    }
+
+    void tokenizePayPalRequest(FragmentActivity activity, PayPalFlowStartedCallback callback) {
+        PayPalRequest paypalRequest = dropInRequest.getPayPalRequest();
+        if (paypalRequest == null) {
+            paypalRequest = new PayPalVaultRequest();
```

Comment:
Should we instead just require they set the `paypalRequest` instead of defaulting to a vault request?

Reply:
We default to a vault request in iOS Drop-in, so it makes sense to me for Android to be consistent with that.

https://github.com/braintree/braintree-ios-drop-in/blob/201f903ce6f444acb6676349e59eec3eb292f34b/Sources/BraintreeDropIn/BTPaymentSelectionViewController.m#L286

Reply:
Good eye 👍